NuclearData/ceNeutronData/aceDatabase/aceNeutronDatabase_class.f90

```diff
@@ -645,6 +707,68 @@ subroutine init_urr(self)
 
   end subroutine init_urr
 
+  !!
+  !!  Checks through all nuclides, creates map with nuclides present and corresponding 0K nuclide
+  !!
+  !!  NOTE: compares the first 5 letters of the ZAID.TT. It would be wrong with isotopes
+  !!        with Z > 99
+  !!
+  subroutine init_DBRC(self, DBRC_nucs, nucSet, map)
+    class(aceNeutronDatabase), intent(inout)     :: self
+    character(nameLen), dimension(:), intent(in) :: DBRC_nucs
+    type(charMap), intent(in)                    :: nucSet
+    type(intMap), intent(out)                    :: map
+    integer(shortInt)                            :: i, j, k, idx
+    character(nameLen)                           :: nuc0K, nucDBRC
+    character(5)                                 :: nuc0Ktemp, nucDBRCtemp
+
+    idx = 1
+
+    ! Loop through DBRC nuclides
+    do i = 1, size(DBRC_nucs)
+      ! Get ZAIDs with and without 0K temperature codes
+      nuc0K = DBRC_nucs(i)
+      nuc0Ktemp = nuc0K(1:5)
+
+      ! Loop through nucSet to find the nucIdxs of the 0K DBRC nuclides
+      k = nucSet % begin()
+      do while (k /= nucSet % end())
+
+        ! Exit loop if a match is found
+        if (nucSet % atKey(k) == nuc0K) then
+          idx = nucSet % atVal(k)
+          exit
+        end if
+
+        ! Increment index
+        k = nucSet % next(k)
+
+      end do
```

Comment:
I think this linear search could be replaced with a get procedure, right? 
We just want to find the value at the key which is the nuclide ZAID?

Reply:
Yes, I don't know how this survived in there! Good point.